### PR TITLE
fix: migrate callback

### PR DESCRIPTION
**Pull request by markusahlstrand**
Migrated the callback. Had to make new versions of some token helpers to not migrate other endpoints at the same time.

### CODE REVIEW

File: src/helpers/apply-token-response-new.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import {
+  AuthParams,
+  AuthorizationResponseMode,
+  CodeResponse,
+  TokenResponse,
+} from "../types";
+
+function applyTokenResponseAsQuery(
```
Comment (markusahlstrand) [inline src/helpers/apply-token-response-new.ts:8]: This is more or less a copy of the old version, but it returns a response object instead

File: src/authentication-flows/social.ts [hunk 2]
```diff
@@ -8,10 +8,10 @@ import {
   LoginState,
 } from "../types";
 import { headers } from "../constants";
-import { setSilentAuthCookies } from "../helpers/silent-auth-cookie";
+import { setSilentAuthCookies } from "../helpers/silent-auth-cookie-new";
 import { generateAuthResponse } from "../helpers/generate-auth-response";
 import { parseJwt } from "../utils/parse-jwt";
-import { applyTokenResponse } from "../helpers/apply-token-response";
+import { applyTokenResponse } from "../helpers/apply-token-response-new";
```
Comment (danielgent) [inline src/authentication-flows/social.ts:14]: :smile: the names explicit enough that we'll remember :+1:

File: src/helpers/apply-token-response-new.ts (hunk 1 above)
Comment (danielgent) [reply src/helpers/apply-token-response-new.ts:8]: _"More or less"_ scares me :see_no_evil:

File: src/routes/tsoa/callback.ts [hunk 3]
```diff
@@ -1,146 +1,181 @@
...
-
-      if (!redirect_uri) {
-        throw new Error("Redirect uri not found");
+import { OpenAPIHono, createRoute, z } from "@hono/zod-openapi";
+import { Var } from "../../types/Var";
+
+export const callback = new OpenAPIHono<{ Bindings: Env; Variables: Var }>()
+  // --------------------------------
+  // GET /.well-known/jwks.json
+  // --------------------------------
```
Comment (danielgent) [inline src/routes/tsoa/callback.ts:12]: Why does every route have this comment? :smile: 

This is a reason not to use comments! :sunglasses:

File: src/routes/tsoa/callback.ts [hunk 4]
```diff
@@ -1,146 +1,181 @@
...
+
+      if (code) {
+        return socialAuthCallback({
+          ctx,
+          state: loginState,
+          code,
+        });
+      }
+
+      throw new HTTPException(400, { message: "State and code are required" });
```
Comment (danielgent) [inline src/routes/tsoa/callback.ts:179]: this all looks copy-pasted from the GET, and doesn't resemble the POST. Maybe you've made other changes, we'll find out soon enough :smile:

**Review by danielgent — APPROVED** [review]

**Pull request merged**